Refactor notification manager and triggerUi to use extension platform

Gudahtt opened the pull request:
The notification manager has been refactored to use the extension platform module instead of using `extensionizer` directly. The extension platform API presents a more ergonomic API, and it correctly
handles errors (which the old notification manager did not). Methods that the extension platform lacked have been added.

It has been updated to use `async/await` instead of callbacks as well, for readability.

The `triggerUI` function has also been updated to use the extension platform instead of `extensionizer`.

Gudahtt committed:
Refactor notification manager to use extension platform

The notification manager has been refactored to use the extension
platform module instead of using `extensionizer` directly. The
extension platform API presents a more ergonomic API, and it correctly
handles errors (which the old notification manager did not). Methods
that the extension platform lacked have been added.

It has been updated to use `async/await` instead of callbacks as well,
for readability.

The `triggerUI` function has also been updated to use the extension
platform instead of `extensionizer`.

Gudahtt commented:
This depends upon #8314

whymarrh commented on app/scripts/lib/notification-manager.js:
```diff
@@ -1,8 +1,10 @@
 const NOTIFICATION_HEIGHT = 620
 const NOTIFICATION_WIDTH = 360
 
+const platform = new ExtensionPlatform()
```
Should this be in the constructor?

whymarrh replied:
One downside of having it in module scope is that it's eagerly created?

Gudahtt replied:
That module is just a collection of functions; it doesn't have any state. It shouldn't be a class in the first place really.

I was considering making it no longer a class in a different PR. That would make it easier to get rid of the `platform` global. I can't think of a reason for it to be a class 🤔

Gudahtt replied:
I guess it couldn't hurt to move it into the constructor in the meantime though. I can do that.

Gudahtt committed:
Move extension platform instantiation into constructor

The platform is now accessible via the `platform` instance variable
within the notification manager.

whymarrh reviewed (approved):
LGTM!

rekmarks reviewed (approved):
LGTM!
